### PR TITLE
MODDCB-52: Secure setup of system users by default

### PR DESCRIPTION
**Pull request by DmytroBykov1**
## Purpose
[MODDCB-52](https://issues.folio.org/browse/MODDCB-52): Secure setup of system users by default

## Approach
In Jenkinsfile the healthChk must be disabled. Jenkins cannot set SYSTEM_USER_PASSWORD env variable. The Jenkins healthChk should be replaced with a maven-failsafe-plugin integration test

#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -7,8 +7,7 @@ buildMvn {
   doDocker = {
     buildJavaDocker {
       publishMaster = true
-      healthChk = true
-      healthChkCmd = 'wget --no-verbose --tries=1 --spider http://localhost:8081/admin/health || exit 1'
+      // healthChk is in FolioDcbApplicationIntegrationTest.java
```
Comment (steveellis) [inline Jenkinsfile:10]: Let's be careful about disabling the health check here. Need to understand what the consequences will be for the community builds.

File: pom.xml [hunk 2]
```diff
@@ -331,6 +344,19 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <version>3.0.0-M6</version>
```
[the remark this answers is not in the document]
Comment (DmytroBykov1) [reply pom.xml]: done

File: pom.xml [hunk 3]
```diff
@@ -130,6 +133,11 @@
       <version>${mapstruct.version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>io.vertx</groupId>
+      <artifactId>vertx-core</artifactId>
+      <version>${vertx.version}</version>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (DmytroBykov1) [reply pom.xml]: <scope>test</scope> added
done

File: src/test/java/org/folio/dcb/FolioDcbApplicationIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+import com.github.tomakehurst.wiremock.junit.WireMockClassRule;
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+import io.vertx.core.json.JsonObject;
+import java.nio.file.Path;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
```
[the remark this answers is not in the document]
Comment (DmytroBykov1) [reply src/test/java/org/folio/dcb/FolioDcbApplicationIntegrationTest.java]: the test class has been completely rebuilt. currently, it's based on JUnit 5 and is triggered by `maven-failsafe-plugin` 
see
https://jenkins-aws.indexdata.com/job/folio-org/job/mod-dcb/job/MODDCB-52/4/consoleFull

File: src/test/java/org/folio/dcb/FolioDcbApplicationIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+public class FolioDcbApplicationIntegrationTest {
```
[the remark this answers is not in the document]
Comment (DmytroBykov1) [reply src/test/java/org/folio/dcb/FolioDcbApplicationIntegrationTest.java]: the test class has been completely rebuilt. currently, it's based on JUnit 5 and is triggered by `maven-failsafe-plugin` 
see
https://jenkins-aws.indexdata.com/job/folio-org/job/mod-dcb/job/MODDCB-52/4/consoleFull

File: Jenkinsfile (hunk 1 above)
Comment (steveellis) [reply Jenkinsfile:10]: Checked with @funkymalc  and removing the healthcheck from the Jenkinsfile is fine.

**Review by gurleenkaurbp — APPROVED** [review]

**Pull request merged**